modules/skybrowser/src/browser.cpp

```diff
@@ -0,0 +1,219 @@
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/skybrowser/include/browser.h>
+
+#include <modules/webbrowser/webbrowsermodule.h>
+#include <modules/webbrowser/include/webkeyboardhandler.h>
+#include <modules/webbrowser/include/browserinstance.h>
+#include <openspace/engine/windowdelegate.h>
+#include <openspace/engine/globals.h>
+#include <openspace/engine/moduleengine.h>
+#include <ghoul/opengl/texture.h>
+#include <ghoul/logging/logmanager.h>
+
+namespace {
+    constexpr const char* _loggerCat = "Browser";
+
+    const openspace::properties::Property::PropertyInfo DimensionsInfo = {
+        "Dimensions",
+        "Browser Dimensions",
+        "Set the dimensions of the web browser window."
+    };
+
+    const openspace::properties::Property::PropertyInfo UrlInfo = {
+        "Url",
+        "URL",
+        "The URL to load"
+    };
+
+    const openspace::properties::Property::PropertyInfo ReloadInfo = {
+        "Reload",
+        "Reload",
+        "Reload the web browser"
+    };
+
+    struct [[codegen::Dictionary(Browser)]] Parameters {
+        // [[codegen::verbatim(DimensionsInfo.description)]]
+        std::optional<glm::vec2> dimensions;
+
+        // [[codegen::verbatim(UrlInfo.description)]]
+        std::optional<std::string> url;
+
+        // [[codegen::verbatim(ReloadInfo.description)]]
+        std::optional<bool> reload;
+    };
+
+#include "browser_codegen.cpp"
+
+} // namespace
+
+namespace openspace {
+
+void Browser::RenderHandler::draw() {}
+
+void Browser::RenderHandler::render() {}
+
+void Browser::RenderHandler::setTexture(GLuint t) {
+    _texture = t;
+}
+
+Browser::Browser(const ghoul::Dictionary& dictionary)
+    : _url(UrlInfo)
+    , _browserPixeldimensions(DimensionsInfo, glm::vec2(500.f), glm::vec2(10.f), glm::vec2(3000.f))
+    , _reload(ReloadInfo)
+{ 
+    if (dictionary.hasValue<std::string>(UrlInfo.identifier)) {
```

Comment:
This entire if-statement can be removed? since you have the `_url = p.url.value_or(_url);` on line 92. That line does exactly what this if-statement does.